ts/features/messages/store/reducers/detailsById.ts

```diff
@@ -50,6 +56,29 @@ export const detailsByIdReducer = (
         [action.payload.id]: pot.noneError(error.message || "UNKNOWN")
       };
     }
+    case getType(populateStoresWithEphemeralAarMessageData): {
+      const { iun, markDown, pnServiceID, subject } = action.payload;
+      const messageData: UIMessageDetails = {
+        hasRemoteContent: true,
+        hasThirdPartyData: true,
+        id: iun,
+        markdown: markDown,
+        serviceId: pnServiceID,
+        subject
+      };
+      return {
+        ...state,
+        [iun]: pot.some(messageData)
+      };
+    }
+    case getType(terminateAarFlow):
+      const newState = _.pickBy(state, (value, _key) =>
+        pipe(
```

Comment:
Instead of piping, what about
`pot.isSome(value) && !isAarDetailById(value.value)`